pkg/store/cas/cas.go

```diff
@@ -11,49 +11,64 @@ import (
 	"fmt"
 
 	ariesstorage "github.com/hyperledger/aries-framework-go/spi/storage"
-	"github.com/ipfs/go-cid"
+	gocid "github.com/ipfs/go-cid"
+	"github.com/ipfs/go-merkledag"
+	"github.com/ipfs/go-unixfs"
 	mh "github.com/multiformats/go-multihash"
 )
 
 // ErrContentNotFound is used to indicate that content as a given address could not be found.
 var ErrContentNotFound = errors.New("content not found")
 
 // CAS represents a content-addressable storage provider.
+// TODO (#344) Support writing and reading both v0 and v1 CIDs.
 type CAS struct {
-	cas ariesstorage.Store
+	cas       ariesstorage.Store
+	useV0CIDs bool // TODO (#344) remove this temporary flag
 }
 
 // New returns a new CAS that uses the passed in provider as a backing store.
-func New(provider ariesstorage.Provider) (*CAS, error) {
+func New(provider ariesstorage.Provider, useV0CIDs bool) (*CAS, error) {
 	cas, err := provider.OpenStore("cas_store")
 	if err != nil {
 		return nil, fmt.Errorf("failed to open store in underlying storage provider: %w", err)
 	}
 
-	return &CAS{cas: cas}, nil
+	return &CAS{cas: cas, useV0CIDs: useV0CIDs}, nil
 }
 
 // Write writes the given content to the underlying storage provider.
 // Returns the address of the content.
 func (p *CAS) Write(content []byte) (string, error) {
-	// TODO #318 figure out why the CIDs produced here differ from the ones that IPFS generates.
-	prefix := cid.Prefix{
-		Version:  0,
-		MhType:   mh.SHA2_256,
-		MhLength: -1, // default length
-	}
+	var cid string
+	if p.useV0CIDs {
+		// The v0 CID produced below is equal to what an IPFS node would produce, assuming that:
+		// 1. The IPFS node is running with default settings, and
+		// 2. The size of the content passed in here is less than 256KB (the default chunk size).
```

Comment:
Is there a plan to make to work with content > 256KB? If there is a plan maybe we should create an issue to track this.

Reply:
Created an issue: #418